ArchiSteamFarm/Steam/Data/InventoryDescription.cs

```diff
@@ -211,6 +211,53 @@ public EAssetRarity Rarity {
 
 								CachedRarity = EAssetRarity.Unknown;
 
+								return CachedRarity.Value;
+						}
+					case "RARITY":
+						string internalName = tag.internal_name.Split('_').Skip(1).FirstOrDefault() ?? tag.internal_name;
+
+						switch (internalName.ToUpperInvariant()) {
+							case "COMMON":
+								CachedRarity = EAssetRarity.Common;
+
+								return CachedRarity.Value;
+							case "UNCOMMON":
+								CachedRarity = EAssetRarity.Uncommon;
+
+								return CachedRarity.Value;
+							case "RARE":
+								CachedRarity = EAssetRarity.Rare;
+
+								return CachedRarity.Value;
+							case "MYTHICAL" or "EPIC":
+								CachedRarity = EAssetRarity.Mythical;
+
+								return CachedRarity.Value;
+							case "LEGENDARY" or "EXOTIC":
+								CachedRarity = EAssetRarity.Legendary;
+
+								return CachedRarity.Value;
+							case "ANCIENT" or "EXTRAORDINARY":
+								CachedRarity = EAssetRarity.Ancient;
+
+								return CachedRarity.Value;
+							case "IMMORTAL" or "CONTRABAND":
+								CachedRarity = EAssetRarity.Immortal;
+
+								return CachedRarity.Value;
+							case "ARCANA":
+								CachedRarity = EAssetRarity.Arcana;
+
+								return CachedRarity.Value;
+							case "UNUSUAL":
+								CachedRarity = EAssetRarity.Unusual;
+
+								return CachedRarity.Value;
+							default:
+								ASF.ArchiLogger.LogGenericError(Strings.FormatWarningUnknownValuePleaseReport(nameof(tag.internal_name), tag.internal_name));
```

Comment:
Should it really report the unknown value considering that rarity names are arbitrary?

Reply:
We can leave it as it is, since we're in charge of doing the parsing, we *do* want to support all possible cases, at least those that make sense.

If there will be some game that'll go entirely out of their minds with rarities like `asdfGABENqwerty` then we'll just tell people to send PR if they want it supported - adding another case to existing switch isn't going to cost us much overhead.